cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/heartbeat/HeartbeatPublisher.java

```diff
@@ -30,6 +32,7 @@
    * @param heartbeat heartbeat to publish
    * @return a {@link ListenableFuture} describing the state of publishing. The {@link ListenableFuture#get} method
    * will return the published heartbeat.
+   * @throws IOException when the {@code heartbeat}
    */
-  ListenableFuture<StreamWriterHeartbeat> sendHeartbeat(StreamWriterHeartbeat heartbeat);
+  ListenableFuture<StreamWriterHeartbeat> sendHeartbeat(StreamWriterHeartbeat heartbeat) throws IOException;
```

Comment:
Oops looks like I didn't finish to write the javadoc for the exception. I added it to cover the fact that publishing a heartbeat as a notification could fail. But now that you mention it, if the exception is thrown in ``DistributedStreamService.runOneIteration``, then the scheduling executor will stop, so I should deal with the exception at that moment.